main.tf

```diff
@@ -0,0 +1,74 @@
+data "aws_iam_policy_document" "lambda_trust_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "lambda" {
+  name               = "${var.lambda_name}"
+  assume_role_policy = "${data.aws_iam_policy_document.lambda_trust_policy.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_basic_role" {
+  role       = "${aws_iam_role.lambda.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_eni_management" {
+  role       = "${aws_iam_role.lambda.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaENIManagementAccess"
+}
+
+resource "aws_iam_role_policy" "lambda" {
+  name   = "${var.lambda_name}"
+  role   = "${aws_iam_role.lambda.name}"
+  policy = "${var.iam_policy_document}"
+  count  = "${var.require_additional_policy}"
+}
+
+resource "aws_lambda_function" "lambda" {
+  s3_bucket     = "${var.lambda_code_bucket}"
+  s3_key        = "${var.lambda_code_path}"
+  function_name = "${var.lambda_name}"
+  role          = "${aws_iam_role.lambda.arn}"
+  runtime       = "${var.lambda_runtime}"
+  handler       = "${var.lambda_handler}"
+  memory_size   = "${var.lambda_memory_size}"
+  timeout       = "${var.lambda_timeout}"
+  tags          = "${merge(var.tags, map("ManagedBy", "Terraform"))}"
+
+  vpc_config {
+    subnet_ids         = ["${var.subnet_ids}"]
+    security_group_ids = ["${var.security_group_ids}"]
+  }
+
+  environment = {
+    variables = "${merge(var.environment_variables, map("ManagedBy", "Terraform"))}"
```

Comment:
Do we still need ManagedBy=Terraform tag? Since all of our resources will be managed by Terraform anyway

Reply:
As confirmed by Sergei, we will still ManagedBy=Terraform reserved for the future, so please ignore this